packages/ai/core/generate-text/generate-text.ts

```diff
@@ -766,12 +756,16 @@ class DefaultGenerateTextResult<TOOLS extends ToolSet, OUTPUT>
     this.steps = options.steps;
     this.providerMetadata = options.providerMetadata;
     this.logprobs = options.logprobs;
-    this.outputResolver = options.outputResolver;
+    this.resolvedOutput = options.resolvedOutput;
     this.sources = options.sources;
   }
 
   get experimental_output() {
-    return this.outputResolver();
+    if (this.resolvedOutput == null) {
+      throw new NoOutputSpecifiedError();
+    }
```

Comment:
`null` output might be valid in some cases. We can remove the error imo

Reply:
(this was the original reason for delaying the check)

Reply:
removing the handling of falsy values leads to a type error. The `generateText()` method could in theory end up returning `null` or `undefined`. Shall we update the type?

```
packages/ai/core/generate-text/generate-text.ts:243:3 - error TS2322: Type 'DefaultGenerateTextResult<TOOLS, Awaited<OUTPUT> | undefined>' is not assignable to type 'GenerateTextResult<TOOLS, OUTPUT>'.
  Types of property 'experimental_output' are incompatible.
    Type 'Awaited<OUTPUT> | undefined' is not assignable to type 'OUTPUT'.
      'OUTPUT' could be instantiated with an arbitrary type which could be unrelated to 'Awaited<OUTPUT> | undefined'.

243   return recordSpan({
      ~~~~~~
```

Reply:
let's create a ticket to revisit this later